src/contrib/cluster/Akka.Cluster.Sharding/PersistentShardCoordinator.cs

```diff
@@ -468,7 +468,8 @@ private void HandleRebalanceDone(RebalanceDone done)
 
             IActorRef region;
             // The shard could have been removed by ShardRegionTerminated
-            if (_currentState.Shards.ContainsKey(done.Shard))
+
+            if (_currentState.Shards.TryGetValue(done.Shard, out region))
```

Comment:
I think this style have a "scoping problem" and can make a code a bit hard to read?
I can probably use it if the variable in question isn't used outside the scope?